report/4.Solution/ParserImplementation.tex

```diff
@@ -115,7 +118,83 @@ \subsection{Implementing a recursive decent parser}
 The method then checks whether the new current symbol is an equals-symbol - if that's the case the declaration is followed by an expression - which means it's an declaration with an assignment otherwise the expression is set to null. 
 The method then returns a Declaration, which is a subclass of AstNode.  
 
+A more complex parsing routine is when an expression is to be parsed. 
+Since an expression in HCL may be either a functioncall or any kind of value, a lot of possibilities arise. 
```

Comment:
Pretty sure function call is two words.
I am also not a huge fan of "a lot of possibilites arise", I will think on it, don't have a good alternative at the moment.